opal/mca/accelerator/base/accelerator_base_null_component.c

```diff
@@ -0,0 +1,248 @@
+static int accelerator_null_close_handle(void *dev_ptr);
+
+static int accelerator_null_host_register(void *ptr, size_t size);
+static int accelerator_null_host_unregister(void *ptr);
+
+static int accelerator_null_get_device(int *dev_id);
+static int accelerator_null_device_can_access_peer( int *access, int dev1, int dev2);
+
+/*
+ * Instantiate the public struct with all of our public information
+ * and pointers to our public functions in it
+ */
+
+opal_accelerator_null_component_t mca_accelerator_null_component = {{
+
+    /* First, the mca_component_t struct containing meta information
+       about the component itself */
+
+    .base_version =
+        {
+            /* Indicate that we are a accelerator v1.1.0 component (which also
+               implies a specific MCA version) */
+
+            OPAL_ACCELERATOR_BASE_VERSION_1_0_0,
+
+            /* Component name and version */
+
+            .mca_component_name = "null",
+            MCA_BASE_MAKE_VERSION(component, OPAL_MAJOR_VERSION, OPAL_MINOR_VERSION,
+                                  OPAL_RELEASE_VERSION),
+
+            /* Component open and close functions */
+
+            .mca_open_component = accelerator_null_open,
+            .mca_close_component = accelerator_null_close,
+            .mca_register_component_params = accelerator_null_component_register,
+
+        },
+    /* Next the MCA v1.0.0 component meta data */
+    .base_data =
+        {/* The component is checkpoint ready */
+         MCA_BASE_METADATA_PARAM_CHECKPOINT},
+    .accelerator_init = accelerator_null_init,
+    .accelerator_finalize = accelerator_null_finalize,
+}};
+
+opal_accelerator_base_module_t opal_accelerator_null_module =
+{
+    accelerator_null_check_addr,
+    accelerator_null_memcpy,
+    accelerator_null_vector_memcpy,
+    accelerator_null_memmove,
+    accelerator_null_malloc,
+    accelerator_null_free,
+    accelerator_null_get_address_range,
+
+    accelerator_null_is_ipc_enabled,
+    accelerator_null_get_handle_size,
+    accelerator_null_get_handle,
+    accelerator_null_open_handle,
+    accelerator_null_close_handle,
+
+    accelerator_null_host_register,
+    accelerator_null_host_unregister,
+
+    accelerator_null_get_device,
+    accelerator_null_device_can_access_peer
+};
+
+static int accelerator_null_open(void)
+{
+    return OPAL_SUCCESS;
+}
+
+static int accelerator_null_close(void)
+{
+    return OPAL_SUCCESS;
+}
+
+static int accelerator_null_component_register(void)
+{
+    return OPAL_SUCCESS;
+}
+
+static opal_accelerator_base_module_t* accelerator_null_init(void)
+{
+    return &opal_accelerator_null_module;
+}
+
+static void accelerator_null_finalize(void)
+{
+    return;
+}
+
+/* Accelerator API's Implementation */
+static int accelerator_null_check_addr(const void *addr, int *dev_id, uint64_t *flags)
+{
+    opal_output(0, "Error: Called method on accelerator_null_check_addr() on Null Accelerator");
```

Comment:
I don't think this is right; This is the function that should work and just always return `0` (not accelerator memory).  Otherwise, what is the point of this component?  Every caller would have to have error handling in the critical path, which is not an improvement over #ifs.

Reply:
yes, we talked about this and I forgot.  Updated.